Mathlib/Data/List/Basic.lean

```diff
@@ -1287,4 +1288,22 @@ lemma left_le_of_mem_range' {a b s x : ℕ} (hx : x ∈ List.range' a b s) : a 
 
 end range'
 
+/-! ### Split -/
+section split
+
+/-- Any `x ∈ l` gives a decomposition `l = l₁ ++ x :: l₂`. -/
+lemma exists_mem_split {l : List α} {x : α} (h : x ∈ l) :
```

Comment:
You can drop this lemma: as I discovered by using `exact?`, it already exists as (one direction of) [List.mem_iff_append](https://leanprover-community.github.io/mathlib4_docs/Init/Data/List/Lemmas.html#List.mem_iff_append).